File: core/bootstrap/src/integrationTest/resources/suite/adhoc/0001.hs
local log = require('log')
local sys = require('sys')
local test = require('test')

-- Remember: Each test is an adhoc invocation
local exec_id = sys.adhoc({
    inputs = {},
    code = [[
        1 + 1
    ]]
})
log.info(exec_id)

local execs = sys.exec.list()
log.info(execs)
test.assert(execs.length == 2)

local exec = execs[1]
test.assert(exec.id == exec_id)

test.complete()

File: faas/mono/src/integrationTest/resources/suite/adhoc/0001.lua
local sys = require('sys')

local err, res = sys.adhoc({
    inputs = {},
    code = [[
        print("inner code execution")
    ]]
})

sys.await_completed(res)

assert(err == nil)
local exec_id = res.id

-- Remember: Each test is an adhoc invocation
local err, execs = sys.exec.list()
assert(err == nil)
assert(#execs == 2)

local exec = execs[1]
assert(exec.id == exec_id)